### PR TITLE
deleteContext fix

### PR DESCRIPTION
**Pull request by ChanochShayner**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/34120

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by ChanochShayner** [comment]
Delete TPB - `playbook-Detonate_File_-_Generic_Test-3.6.0.yml` and replace the use of it in the new according to - https://panw-global.slack.com/archives/G011E63JXPB/p1616661160115600

**Comment by ChanochShayner** [comment]
In this script all the information is deleted from the context, if we use the KeysToKeep argument then we first save the information, delete everything from the context, and then return the requested information back.
My improvement -
Because there is a problem returning DBotScore in entry - if there is DBotScore in KeysToKeep we do not delete it from context at all.
@guykeller FYI

File: Packs/CommonScripts/Scripts/script-DeleteConext.yml [hunk 1]
```diff
@@ -63,9 +63,11 @@ script: >
           if (isSubPlaybookKey) {
               key = 'subplaybook-${currentPlaybookID}.' + key;
           }
-          var result = executeCommand('delContext', {key: key});
-          if(!result || result.type === entryTypes.error) {
-              errorsStr = errorsStr + "\n" + result.Contents;
+          if (key !== "DBotScore" || !DBotScore) {
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/script-DeleteConext.yml:66]: what is `DBotScore`? where did you set it?

**Review by guykeller — COMMENTED** [review]
Did you test this? If so, how? Can we see the impact somehow?

File: Packs/CommonScripts/Scripts/script-DeleteConext.yml (hunk 1 above)
Comment (ChanochShayner) [reply Packs/CommonScripts/Scripts/script-DeleteConext.yml:71]: my bad, done.

**Comment by ChanochShayner** [comment]
@guykeller If I ran `!ip ip=11.11.11.11` and `!ip ip=22.22.22.22` 
Before fixing `deleteContext` if running `!DeleteContext all=yes keysToKeep=DBotScore,IP`
The indicator insightCash would get the information of the other indicator (both, under IP and under DBotScore)
After the fix we see that it's ok, 
attached pictures of before and after the fix -

DBotScore Before - 
image

Indicator Before - 
image

DBotScore After - 
image

Indicator After - 
image
@MarkShleifer FYI

**Review by BenJoParadise — COMMENTED** [review]
The code change looks right to me, but not sure how content PR approval cycle goes nowadays, so i'll leave the final approval to someone from content.

**Comment by BenJoParadise** [comment]
Also needs to resolve conflicts in release notes, apparently.

File: Packs/CommonScripts/Scripts/script-DeleteConext.yml [hunk 2]
```diff
@@ -43,6 +43,12 @@ script: >
   if (shouldDeleteAll) {
       var keysToKeep = (args.keysToKeep) ? args.keysToKeep.split(',').map(function(item) { return item.trim(); }) : [];
       var keysToKeepObj = {};
+      var DBotScore = false;
```
Comment (guykeller) [inline Packs/CommonScripts/Scripts/script-DeleteConext.yml:46]: Let's give a better name to this variable

**Review by guykeller — CHANGES_REQUESTED** [review]

File: Packs/CommonScripts/Scripts/script-DeleteConext.yml [hunk 3]
```diff
@@ -63,9 +69,11 @@ script: >
           if (isSubPlaybookKey) {
               key = 'subplaybook-${currentPlaybookID}.' + key;
           }
-          var result = executeCommand('delContext', {key: key});
-          if(!result || result.type === entryTypes.error) {
-              errorsStr = errorsStr + "\n" + result.Contents;
+          if (key !== "DBotScore" || !DBotScore) {
```
Comment (guykeller) [inline Packs/CommonScripts/Scripts/script-DeleteConext.yml:72]: When would the key be equal to DBotScore? Don't we delete it if it's there?

Comment (ChanochShayner) [reply Packs/CommonScripts/Scripts/script-DeleteConext.yml:72]: We delete it from `keysToKeep`, this loop is on `fieldsToDelete`

File: Packs/CommonScripts/Scripts/script-DeleteConext.yml [hunk 4]
```diff
@@ -43,6 +43,12 @@ script: >
   if (shouldDeleteAll) {
       var keysToKeep = (args.keysToKeep) ? args.keysToKeep.split(',').map(function(item) { return item.trim(); }) : [];
       var keysToKeepObj = {};
+      var DBotScore = false;
+      index = keysToKeep.indexOf("DBotScore");
+      if (index > -1) {
+        keysToKeep.splice(index, 1);
+        DBotScore = true
```
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply Packs/CommonScripts/Scripts/script-DeleteConext.yml:50]: done

File: Packs/CommonScripts/Scripts/script-DeleteConext.yml (hunk 2 above)
Comment (ChanochShayner) [reply Packs/CommonScripts/Scripts/script-DeleteConext.yml:46]: done

**Review by guykeller — APPROVED** [review]

**Comment by guykeller** [comment]
Approved, but it seemed like the build failed before. Try to see why

File: Packs/CommonScripts/ReleaseNotes/1_3_32.md [hunk 5]
```diff
@@ -0,0 +1,4 @@
+
+#### Scripts
+##### DeleteContext
+- Fixed an issue where unrelated data stored in indicator InsightCache when using **KeysToKeep** to keep DBotScore data in the context.
```
Comment (amshamah419) [inline Packs/CommonScripts/ReleaseNotes/1_3_32.md:4]:
```suggestion
- Fixed an issue where unrelated data was stored in the indicator InsightCache while using **KeysToKeep** in order to keep DBotScore data in the context.
```

Comment (ChanochShayner) [reply Packs/CommonScripts/ReleaseNotes/1_3_32.md]: done

**Pull request merged**